src/token.jl

```diff
@@ -95,6 +95,204 @@ function kind(t::AbstractToken)
     return t.kind
 end
 exactkind(t::AbstractToken) = t.kind
+
+Meta.parse(t::T) where {T <: Union{Token, Array{Token}}} = Meta.parse(untokenize(t))
+
+"""
+    UndefToken
+
+An abstract type that shows if the an evaluated `Token` is defined.
+"""
+abstract type UndefToken end
```

Comment:
This should probably be a concrete type `struct UndefToken` and then you return `UndefToken()`, similar to how `nothing == Nothing()`.